server/sockets/cohort.js

```diff
@@ -187,136 +254,311 @@ const updateCohortHeaderStatus = (socket, cohortClientLocks) => {
+const deleteCohort = (io, allCohortsClients) => data => {
+  const { cohortId, owners } = data;
 
-      socket.broadcast
+  io.sockets
+    .to(cohortChannel(cohortId))
+    .emit(CohortActionTypes.REMOVE_WHEN_COHORT_UNAVAILABLE, cohortId);
+
+  owners.forEach(id => {
+    const ownerId = id.toString();
+
+    if (allCohortsClients.has(ownerId)) {
+      const { socketId } = allCohortsClients.get(ownerId);
+
+      io.sockets
         .to(socketId)
         .emit(CohortActionTypes.DELETE_SUCCESS, { cohortId });
     }
   });
+};
 
-const deleteCohort = (socket, allCohortsClients) =>
-  socket.on(CohortActionTypes.DELETE_SUCCESS, data => {
-    const { cohortId, members } = data;
-
-    removeCohort(socket, cohortId, allCohortsClients, members);
-  });
-
-const restoreCohort = (socket, allCohortsClients, cohortClients) =>
-  socket.on(CohortActionTypes.RESTORE_SUCCESS, data => {
-    const { cohortId } = data;
+const restoreCohort = (
+  io,
+  allCohortsClients,
+  cohortClients,
+  dashboardClients
+) => data => {
+  const { cohortId } = data;
 
-    Cohort.findById(cohortId)
-      .populate('memberIds', 'avatarUrl displayName _id')
-      .lean()
-      .exec()
-      .then(doc => {
-        if (doc) {
-          const { memberIds, ownerIds } = doc;
-          const cohortMetaData = responseWithCohort(doc);
+  return Cohort.findById(cohortId)
+    .populate('memberIds', 'avatarUrl displayName _id')
+    .lean()
+    .exec()
+    .then(doc => {
+      if (doc) {
+        const { memberIds, ownerIds } = doc;
+        const cohortMetaData = responseWithCohort(doc);
 
-          ownerIds.forEach(id => {
-            const ownerId = id.toString();
+        ownerIds.forEach(id => {
+          const ownerId = id.toString();
 
-            if (cohortClients.has(ownerId)) {
-              const { socketId } = cohortClients.get(ownerId);
+          if (cohortClients.has(ownerId)) {
+            const { socketId, viewId } = cohortClients.get(ownerId);
 
-              socket.broadcast
+            if (viewId === cohortId) {
+              io.sockets
                 .to(socketId)
                 .emit(
                   CohortActionTypes.FETCH_DETAILS_SUCCESS,
                   responseWithCohortDetails(doc, ownerId)
                 );
             }
-          });
+          }
+        });
+
+        memberIds.forEach(member => {
+          const { _id } = member;
+          const memberId = _id.toString();
+
+          if (allCohortsClients.has(memberId)) {
+            const { socketId } = allCohortsClients.get(memberId);
+
+            io.sockets
+              .to(socketId)
+              .emit(CohortActionTypes.FETCH_META_DATA_SUCCESS, {
+                [cohortId]: cohortMetaData
+              });
+          }
+        });
+
+        return List.find({ cohortId, isArchived: false })
+          .populate('cohortId', 'ownerIds')
+          .lean()
+          .exec();
+      }
+    })
+    .then(docs => {
+      if (docs && docs.length > 0) {
```

Comment:
Only a single list processed? Shouldn't all of them be iterated?

Reply:
`dosc` is an array of documents, if there is at least one doc (`docs && docs.length > 0`) we process this array